### PR TITLE
`@GsonIgnore` annotation

### PR DESCRIPTION
**Pull request by frett**
this moves some of the work @wrandall22 did on Gson for MPDX last sprint to a common reusable library.

### CODE REVIEW

File: gto-support-api/build.gradle [hunk 1]
```diff
@@ -2,6 +2,6 @@ dependencies {
     compile project(':gto-support-core')
     compile project(':gto-support-api-base')
 
-    compile 'com.google.guava:guava:18.0'
+    compile 'com.google.guava:guava:' + deps.guava
```
Comment (dsgoers) [inline gto-support-api/build.gradle]: What does this change do?

Comment (frett) [inline gto-support-api/build.gradle]: allows configuration of dependency version numbers to be central instead of sprinkled throughout the code.

Comment (frett) [inline gto-support-api/build.gradle]: here is where they are defined:

https://github.com/CruGlobal/android-gto-support/blob/master/build.gradle#L13

File: gto-support-gson/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest package="org.ccci.gto.android.common.gson"/>
```
Comment (dsgoers) [inline gto-support-gson/src/main/AndroidManifest.xml]: Still need a manifest even though it's not an actual app?

Comment (frett) [inline gto-support-gson/src/main/AndroidManifest.xml]: yeah, it's still a library project which requires a bundled manifest. Any resources defined in a library project get merged into the main project manifest, but I've only used that in a couple places before.

Comment (frett) [inline gto-support-gson/src/main/AndroidManifest.xml]: I could have made this a jar currently, which wouldn't have had that constraint. But the shared gradle build script is currently setup to build everything as android libraries.

**Comment by frett** [comment]
I prodded travis and it successfully completed the build

**Comment by dsgoers** [comment]
Looks good, great idea @wrandall22 !

**Comment by wrandall22** [comment]
Thanks @dsgoers

**Pull request merged**